index.html

```diff
@@ -919,11 +945,100 @@ <h4>Verify Proof (eddsa-jcs-2022)</h4>
             </dd>
           </dl>
 
-          <p>
-The algorithm is the same as the one described in Section
-[[[#verify-proof-eddsa-rdfc-2022]]] with the changes outlined in
-Section [[[#eddsa-jcs-2022]]].
-          </p>
+          <ol class="algorithm">
+            <li>
+Let |unsecuredDocument| be a copy of |securedDocument| with
+the `proof` value removed.
+            </li>
+            <li>
+Let |proofOptions| be the result of a copy of |securedDocument|.|proof| with `proofValue`
+removed.
+            </li>
+            <li>
+Let |proofBytes| be the
+<a data-cite="VC-DATA-INTEGRITY#multibase-0">Multibase decoded base58-btc
+value</a> in |securedDocument|.|proof|.|proofValue|.
+            </li>
+            <li>
+Perform the following checks  between the |securedDocument|.<var>@context</var> and
+the |proofOptions|.<var>@context</var>:
+              <ol class="algorithm">
+                <li>
+The |securedDocument|.<var>@context</var> MUST contain all values contained in the
+|proofOptions|.<var>@context</var> in the same order. Otherwise, set |verified| to
+`false` and skip to the last step.
+                </li>
+                <li>
+The |securedDocument|.<var>@context</var> MUST NOT have any values preceding the
+values contained in the |proofOptions|.<var>@context</var>, but those MAY be followed by
+additional values not contained in the |proof|.<var>@context</var>. Otherwise,
+set |verified| to `false` and skip to the last step.
+                </li>
+              </ol>
+
+            </li>
+            <li>
+Set |securedDocument|.<var>@context</var> equal to
+|proofOptions|.<var>@context</var>.
```

Comment:
Yes, that is an important point: in this case, we need to assume that we're in 'compatibility mode' for support of proofs generated according to the specification as it existed prior to this change, and _NOT_ touch the document's @context at all

Reply:
Hi all (@rflechtner, @silverpill, @dlongley, @msporny ), in making this update, for simplicity and security, I put back step 8 from [Proof Configuration (eddsa-rdfc-2022)](https://w3c.github.io/vc-di-eddsa/#proof-configuration-eddsa-rdfc-2022) which copies the existing document `@context` into *proofConfig* prior to canonicalization and then hashing.  This further protects the contents of the *proof* field, but also needs to be accounted for to realize backward compatibility (results in different hash values being signed).

It seems that all algorithm tweaks can be detected and compensated for via the presence/absence of the *proof.@context* field. 

**Discussion**: Do we want to add these complications? How do we "officially" talk (or not) about this backwards compatibility? Would greatly appreciate help on informative text for (a) this approach to handling context with JCS, and (b) backwards compatibility.

Note my background is more from the cryptography side rather than JSON-LD side and I've been generating most of the test vectors for the specs so I'm used to implementing/tweaking the algorithms. Cheers Greg

Reply:
Defining a new cryptosuite might still be worth it, if more normative changes are planned (https://github.com/w3c/vc-data-integrity/issues/231; potential problems with nested secured documents?)

Reply:
> in making this update, for simplicity and security, I put back step 8 from [Proof Configuration (eddsa-rdfc-2022)](https://w3c.github.io/vc-di-eddsa/#proof-configuration-eddsa-rdfc-2022) which copies the existing document `@context` into _proofConfig_ prior to canonicalization and then hashing. This further protects the contents of the _proof_ field, but also needs to be accounted for to realize backward compatibility (results in different hash values being signed).

Thanks for pointing this out, I had missed this detail. Indeed, writing the document context to the proof context during verification would result in a loss of interoperability with existing implementations, which I think was a goal. I'd argue we would thus either have to explicitly disallow step 8 _only on verification_, or bump the cryptosuite version as @silverpill suggested (https://github.com/w3c/vc-di-eddsa/pull/79#discussion_r1587384419), in which case we do not need to consider backwards compatibility. Don't know if the latter is on the table though, as I'm not too involved with the WG. @dlongley & @msporny would be better suited to make a call here.

As for discussing reasons for these steps, I think that would help avoiding implementation errors, as instructions like 'omit step 8' that are not immediately plausible are more easily overlooked.

Reply:
Hi @silverpill and @rflechtner, coming up with a new cryptosuite is a significant burden. To remain "compatible" with the "previous versions" of the procedures involving JCS (EdDSA and ECDSA) would just require a simple check for the *proof.@context* field and some slightly different processing if its not present. This seems the best way to go to me.

That is we would be able to verify previously signed JCS based VC, but we would specify that all (new) VCs based on JCS would only use the new procedures (since those VCs are tolerant to injection, can be used in proof chains/sets, etc...).

Thoughts? If you concur I can enhance the verification procedure appropriately and add a test vector showing this.

Reply:
It seems that if we delete adding the context within the JCS proof configuration algorithm, then we can make these two additional changes:

1. During proof generation, the unsecured document context should be added to the proof options before running the proof configuration algorithm. This will cause the same effect as having the proof configuration algorithm add it.
2. During proof verification, if `@context` is present in the proof options, ensure the secured document `@context` values start with its value and then set the unsecured document `@context` to the value from the proof options. This will cause no `@context` to be present later in the proof configuration algorithm for the backwards compatibility case (because nothing will add it). For the regular case, `@context` will be present because it was in the proof options (coming from the `proof` in the document during verification). If a holder removes `@context` from the proof in the regular case, then it will fail to verify because it was present when signed.

Reply:
Hi @dlongley, we don't have a separate "JCS Proof Configuration" algorithm. So I implemented your above suggestion just by modifying the use of the "RDF Proof Configuration" algorithm in the verification step. In addition, I cleaned up some mistakes where *securedDocument* should have been *unsecuredDocument*. Please review again.

Reply:
Hi all, just updated test vector code with the compatibility checks discussed here. See https://github.com/Wind4Greg/EdDSA-Test-Vectors/blob/jcs-injection/JCSDataIntegrityVerify.js and was able to verify both old and new JCS EdDSA test vectors. Cheers Greg